src/grpc/infer_handler.cc

```diff
@@ -916,8 +916,29 @@ ModelInferHandler::Execute(InferHandler::State* state)
   if (err == nullptr) {
     TRITONSERVER_InferenceTrace* triton_trace = nullptr;
 #ifdef TRITON_ENABLE_TRACING
-    state->trace_ =
-        std::move(trace_manager_->SampleTrace(request.model_name()));
+    TraceStartOptions start_options;
+    InferenceTraceMode mode = TRACE_MODE_TRITON;
+    trace_manager_->GetTraceMode(request.model_name(), &mode);
```

Comment:
Looks like this adds a lock to the hot path. Will DM you to discuss further

Reply:
Technically, I can do the following re-factoring:
`std::shared_ptr<TraceSetting> trace_setting` can become a part of `TraceStartOptions`. 

And then when we initialize `start_options` on line 919, I can get `trace_setting` for current model through [the steps](https://github.com/triton-inference-server/server/blob/main/src/tracer.cc#L296-L301). This way, lock is happenning only once (as it is now) and then we just pass start options to `TraceManager::SampleTrace` , where we move on to [this line](https://github.com/triton-inference-server/server/pull/6785/files#diff-c05667609085c993fd282ff18d0429945c44492b95c0d01fcbeec6946fda2db2R316) since `trace_setting` is already obtained

Reply:
@GuanLuo I would like to have your opinion on this as well if possible

Reply:
discussed offline